### PR TITLE
add assert_addrs_fit

### PR DESCRIPTION
**Pull request by Roger-luo**

### CODE REVIEW

File: src/error.jl [hunk 1]
```diff
@@ -163,8 +163,23 @@ Assert if all the locations are:
...
+    @assert_locs_fit <number of qubits> <locations list> [<msg>]
+
+Assert if the location is exactly fit into n qubits.
+"""
+macro assert_locs_fit(n, locs, msgs...)
+    msg = process_msgs(msgs...; default="locations is not contiguous.")
+    quote
+        @assert_locs_safe $(esc(n)) $(esc(locs))
+        addrs |> first |> minimum == 1 || throw(AddressConflictError("addr not exact fit at 1"))
+        islocs_contiguous($(esc(n)), $(esc(locs))) || error($msg)
```
Comment (GiggleLiu) [inline src/error.jl]: addrs |> last |> maximum == n || throw(AddressConflictError("addr not exact fit at end"))

Comment (Roger-luo) [reply src/error.jl]: I copied this from Yao v0.3.2 why change the order?

Comment (GiggleLiu) [reply src/error.jl]: you need to check both start and end.

Comment (GiggleLiu) [reply src/error.jl]: And, you need to sort the locs first.

Comment (GiggleLiu) [reply src/error.jl]: And please write tests.

**Pull request merged**